lib/SPIRV/SPIRVLowerLLVMIntrinsic.cpp

```diff
@@ -0,0 +1,207 @@
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.i8",          "llvm_bitreverse_i8",          LLVMBitreverseScalar},
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.i16",         "llvm_bitreverse_i16",         LLVMBitreverseScalar},
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.i32",         "llvm_bitreverse_i32",         LLVMBitreverseScalar},
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.i64",         "llvm_bitreverse_i64",         LLVMBitreverseScalar},
+
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.v2i8",        "llvm_bitreverse_v2i8",        LLVMBitreverseV2},
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.v2i16",       "llvm_bitreverse_v2i16",       LLVMBitreverseV2},
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.v2i32",       "llvm_bitreverse_v2i32",       LLVMBitreverseV2},
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.v2i64",       "llvm_bitreverse_v2i64",       LLVMBitreverseV2},
+
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.v4i8",        "llvm_bitreverse_v4i8",        LLVMBitreverseV4},
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.v4i16",       "llvm_bitreverse_v4i16",       LLVMBitreverseV4},
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.v4i32",       "llvm_bitreverse_v4i32",       LLVMBitreverseV4},
+  {Intrinsic::bitreverse,         ExtensionID::SPV_KHR_bit_instructions, "llvm.bitreverse.v4i64",       "llvm_bitreverse_v4i64",       LLVMBitreverseV4},
+
+  {Intrinsic::sadd_with_overflow, NO_SUPPORTING_EXTENSION,               "llvm.sadd.with.overflow.i16", "llvm_sadd_with_overflow_i16", LLVMSaddWithOverflow},
+  {Intrinsic::sadd_with_overflow, NO_SUPPORTING_EXTENSION,               "llvm.sadd.with.overflow.i32", "llvm_sadd_with_overflow_i32", LLVMSaddWithOverflow},
+  {Intrinsic::sadd_with_overflow, NO_SUPPORTING_EXTENSION,               "llvm.sadd.with.overflow.i64", "llvm_sadd_with_overflow_i64", LLVMSaddWithOverflow},
+};
+// clang-format on
+
+} // namespace
+
+void SPIRVLowerLLVMIntrinsicBase::visitIntrinsicInst(CallInst &I) {
+  IntrinsicInst *II = dyn_cast<IntrinsicInst>(&I);
+
+  const LLVMIntrinsicMapEntryType *MapEntry{nullptr};
+
+  if (!II)
+    return;
+
+  Function *IntrinsicFunc = I.getCalledFunction();
+  assert(IntrinsicFunc && "Missing function");
+  StringRef IntrinsicName = IntrinsicFunc->getName();
+
+  for (const LLVMIntrinsicMapEntryType &LLVMIntrinsicMapEntry :
+       LLVMIntrinsicMapEntries) {
+    if (II->getIntrinsicID() == LLVMIntrinsicMapEntry.ID &&
+        IntrinsicName == LLVMIntrinsicMapEntry.LLVMFuncName &&
```

Comment:
There might be a performance issue here.  If I understand correctly, for any intrinsic other than `llvm.bitreverse` and `llvm.sadd.with.overflow`, the pass now has to iterate over all `LLVMIntrinsicMapEntries` entries.

Reply:
Updated to use a std::map